src/alfred/self_command.py

```diff
@@ -19,6 +26,45 @@ def check():
         echo.error("Fail to load some commands")
         raise Exit(code=1)
 
+def new(shell_cmd: Optional[str] = None, project_dir: Optional[str] = None):
+    if project_dir is None:
+        project_dir = manifest.lookup_project_dir()
+
+    if shell_cmd is None:
+        code = 'pass'
+    else:
+        code = f"alfred.run('{shell_cmd}')"
+
+    all_commands = commands.list_all()
+    all_commands_names = sorted([cmd.original_name for cmd in all_commands])
+    new_cmd_name = alfred_prompt.prompt('What command do you want to create ? ', all_commands_names, validation_func=lambda p: _check_cmd_should_be_unique(p, all_commands_names))
+    new_cmd_description = alfred_prompt.prompt(f'What the command `{new_cmd_name}` will do ? ', default="")
+
+    all_modules = commands.list_modules()
+    all_command_directories = commands.list_command_directories()
+    default_command_module = os.path.join(all_command_directories[0], 'commands.py')
+
+    module_target = alfred_prompt.prompt(f'In which module do you want to create the `{new_cmd_name}` command ? ',
+                                     all_modules,
+                                     default=default_command_module,
+                                     validation_func=lambda p: _check_module_is_valid(p, all_command_directories))
+
+    new_cmd_name_slug = sluggify(new_cmd_name)
+    tpl_command_variables = {
+        'cmd': new_cmd_name,
+        'cmd_slug': new_cmd_name_slug,
+        'cmd_description': new_cmd_description,
+        'cmd_module': module_target,
+        'code': code
+    }
+    command_presentation = resource.template('command_presentation.tpl', variables=tpl_command_variables)
+
+    scaffhold = alfred_prompt.prompt(command_presentation.strip('\n'), default='y', validation_func=lambda p: None if p in ['y', 'n'] else 'answer with y or n')
```

Comment:
use the alfred_prompt.confirm function which already has the correct validation primitives.